posthog-android/src/main/java/com/posthog/android/internal/PostHogAndroidUtils.kt

```diff
@@ -144,6 +150,38 @@ internal fun Activity.activityLabelOrName(config: PostHogAndroidConfig): String?
     }
 }
 
+@Suppress("DEPRECATION")
+internal fun Activity.getReferrerInfo(config: PostHogAndroidConfig): Map<String, String>? {
+    return try {
+        val intent = this.intent
+        val referrerInfoMap = mutableMapOf<String, String>()
+        val referrer: Uri?
+        val referrerName: String?
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP_MR1) {
+            referrer = intent.getParcelableExtra(Intent.EXTRA_REFERRER)
+            referrerName = intent.getStringExtra(Intent.EXTRA_REFERRER_NAME)
+        } else {
+            referrer = intent.getParcelableExtra("android.intent.extra.REFERRER")
+            referrerName = intent.getStringExtra("android.intent.extra.REFERRER_NAME")
+        }
+        if (referrer == null && referrerName == null) {
+            return null
+        } else if (referrer == null) {
+            Uri.parse(referrerName)?.let { uri ->
+                referrerInfoMap["\$referrer"] = uri.toString()
+                uri.host?.let { referrerInfoMap["\$referring_domain"] = it }
+            }
+        } else {
+            referrerInfoMap["\$referrer"] = referrer.toString()
+            referrer.host?.let { referrerInfoMap["\$referring_domain"] = it }
+        }
```

Comment:
I think it'd be easier to read if you do an `if` to `referrer` and another `if` to `referrerName` so you don't need the if/elseif/else

Reply:
Okay, I'll change it